Connection Manager: Prevent notice on empty Jetpack blog token

gititon opened the pull request:
This PR prevents a notice when the Jetpack blog token is empty.

Testing instructions:
Before applying this patch, add `define( 'JETPACK_BLOG_TOKEN', '' );` to wp-config.php and ensure Jetpack Comments are enabled. You should receive a notice similar to the following:

```
[03-Jul-2019 20:24:09 UTC] PHP Notice:  Uninitialized string offset: 0 in /var/www/html/wp-content/plugins/jetpack/packages/connection/src/Manager.php on line 615
[03-Jul-2019 20:24:09 UTC] PHP Stack trace:
[03-Jul-2019 20:24:09 UTC] PHP   1. {main}() /var/www/html/index.php:0
[03-Jul-2019 20:24:09 UTC] PHP   2. require() /var/www/html/index.php:17
[03-Jul-2019 20:24:09 UTC] PHP   3. require_once() /var/www/html/wp-blog-header.php:19
[03-Jul-2019 20:24:09 UTC] PHP   4. include() /var/www/html/wp-includes/template-loader.php:78
[03-Jul-2019 20:24:09 UTC] PHP   5. comments_template() /var/www/html/wp-content/themes/twentysixteen/single.php:24
[03-Jul-2019 20:24:09 UTC] PHP   6. require() /var/www/html/wp-includes/comment-template.php:1510
[03-Jul-2019 20:24:09 UTC] PHP   7. comment_form() /var/www/html/wp-content/themes/twentysixteen/comments.php:77
[03-Jul-2019 20:24:09 UTC] PHP   8. do_action() /var/www/html/wp-includes/comment-template.php:2558
[03-Jul-2019 20:24:09 UTC] PHP   9. WP_Hook->do_action() /var/www/html/wp-includes/plugin.php:465
[03-Jul-2019 20:24:09 UTC] PHP  10. WP_Hook->apply_filters() /var/www/html/wp-includes/class-wp-hook.php:310
[03-Jul-2019 20:24:09 UTC] PHP  11. Jetpack_Comments->comment_form_after() /var/www/html/wp-includes/class-wp-hook.php:286
[03-Jul-2019 20:24:09 UTC] PHP  12. Jetpack_Data::get_access_token() /var/www/html/wp-content/plugins/jetpack/modules/comments/comments.php:280
[03-Jul-2019 20:24:09 UTC] PHP  13. Automattic\Jetpack\Connection\Manager->get_access_token() /var/www/html/wp-content/plugins/jetpack/class.jetpack-data.php:11
```

Once this patch is applied, the notice should go away.

gititon committed:
Prevent notice on empty blog token

gititon committed:
update

jeherve reviewed (changes requested):
Tests seem to be failing right now.

I also tried to set a real value for `JETPACK_BLOG_TOKEN` and then I deleted my site's `blog_token` Jetpack option to get into a situation close to some of those Atomic sites. When on Master, that seems to work; if I head over to Tools > Site Health > Info and look at my site's site info, I have a blog token set there; it picks it up from `JETPACK_BLOG_TOKEN`. When I check out this branch, however, no token is set at all.

gititon committed:
fix

gititon committed:
fix

gititon commented:
> Tests seem to be failing right now.
> 
> I also tried to set a real value for `JETPACK_BLOG_TOKEN` and then I deleted my site's `blog_token` Jetpack option to get into a situation close to some of those Atomic sites. When on Master, that seems to work; if I head over to Tools > Site Health > Info and look at my site's site info, I have a blog token set there; it picks it up from `JETPACK_BLOG_TOKEN`. When I check out this branch, however, no token is set at all.

Hi Jeremy,

Using this branch, I
1) copied my Jetpack `Blog Public Key` as reported by Tools -> Site Health -> Info
2) Set it as my JETPACK_BLOG_TOKEN via `define( 'JETPACK_BLOG_TOKEN', '<my-token>' );` in wp-config.php
3) Deleted the option by ssh'ing into my Jetpack sandbox and `wp jetpack options delete blog_token --allow-root`
4) The value of `Blog Public Key` changed to `Potentially Malformed Token.` 

I switched to master, disconnected and reconnected Jetpack, ensured the `blog_token` value was restored in Tools -> Site Health -> Info, and repeated the same steps above. I got the same result, `Potentially Malformed Token.`

jeherve reviewed (approved):
This seems to work well for me. Merging!

jeherve commented:
> I got the same result, Potentially Malformed Token.

Yes, that's because the token you can use as `JETPACK_BLOG_TOKEN` has to be the full token, not just the public part of it.

jeherve commented:
Cherry-picked to `branch-7.5` in f1a8da225b3d66752cc29915650d80459a22fbba
